### ISSUE TITLE
Custom plotting from workers on Bokeh

### ISSUE THREAD

**Issue by CarlosCuetoM** [dask/distributed#2790]
Hi!

I'd be interested in plotting data during task execution in the workers without having to send the data to the client, is there an appropriate way to use the Bokeh server that is already running to do this? Is this the best approach?

Thanks,
Carlos

**Comment by mrocklin** [comment]
Yes, with a small amount of work this should be doable.  You would want to add a new Bokeh application to the Bokeh server living on the worker or scheduler.

Here is how we specify the applications on the scheduler

https://github.com/dask/distributed/blob/eba954b0ce589b2186e7ea78d697a79ac1faad62/distributed/dashboard/scheduler.py#L1596-L1617

and worker

https://github.com/dask/distributed/blob/eba954b0ce589b2186e7ea78d697a79ac1faad62/distributed/dashboard/worker.py#L789-L796

You might make this change in a preload script or a worker plugin

https://docs.dask.org/en/latest/setup/custom-startup.html

The bokeh servers are available with something like `worker.services['dashboard'].server` (or `scheduler.`)

**Comment by CarlosCuetoM** [comment]
That's exactly what I was looking for, thanks. I'll give it a go in the next few weeks!
